### PR TITLE
Enable TreatWarningsAsErrors for some projects

### PR DESCRIPTION
**Pull request by Bond-009**
Analyzers are only run in debug build, so setting `TreatWarningsAsErrors`
for release build will catch the compiler warnings until we resolve all
analyzer warnings.

### CODE REVIEW

File: Emby.Naming/Emby.Naming.csproj [hunk 1]
```diff
@@ -21,13 +25,13 @@
     <RepositoryUrl>https://github.com/jellyfin/jellyfin</RepositoryUrl>
   </PropertyGroup>
 
-  <!-- Code analysers-->
-  <ItemGroup Condition=" '$(Configuration)' == 'Debug' ">
+  <!-- Code Analyzers-->
+  <!-- <ItemGroup Condition=" '$(Configuration)' == 'Debug' ">
```
Comment (EraYaN) [inline Emby.Naming/Emby.Naming.csproj]: Is there a reason to disable the analyzers even in Debug?

**Comment by Bond-009** [comment]
@EraYaN They are only being added to files that aren't documented 100%, this way new files are required to have full documentation.

File: Emby.Server.Implementations/Emby.Server.Implementations.csproj [hunk 2]
```diff
@@ -50,13 +50,13 @@
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
   </PropertyGroup>
 
-  <!-- Code analysers-->
-  <ItemGroup Condition=" '$(Configuration)' == 'Debug' ">
+  <!-- Code Analyzers-->
+  <!--<ItemGroup Condition=" '$(Configuration)' == 'Debug' ">
```
Comment (EraYaN) [inline Emby.Server.Implementations/Emby.Server.Implementations.csproj]: I missed this one where is was disabled as well.

File: Emby.Photos/Emby.Photos.csproj [hunk 3]
```diff
@@ -20,12 +24,12 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
   </PropertyGroup>
 
-  <!-- Code analysers-->
-  <ItemGroup Condition=" '$(Configuration)' == 'Debug' ">
+  <!-- Code Analyzers-->
+  <!--<ItemGroup Condition=" '$(Configuration)' == 'Debug' ">
```
Comment (EraYaN) [inline Emby.Photos/Emby.Photos.csproj]: And here.

**Pull request merged**